### PR TITLE
[DOCS] Overhaul snapshot and restore docs

### PR DESCRIPTION
**Pull request by jrodewig**
Makes several changes to consolidate snapshot and backup-related docs.

Highlights:

* Adds info about supported ESS snapshot repository types
* Adds docs for Kibana's Snapshot and Restore feature
* Combines tutorial pages related to taking and managing snapshots
* Consolidates explanations of the snapshot process
* Incorporates SLM into the snapshot tutorial
* Removes duplicate "back up a cluster" pages

### Previews

* Snapshot and restore overview: https://elasticsearch_79081.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/snapshot-restore.html
* Register a snapshot repo: https://elasticsearch_79081.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/snapshots-register-repository.html
* Take a snapshot: https://elasticsearch_79081.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/snapshots-take-snapshot.html

### CODE REVIEW

**Review by DaveCTurner — COMMENTED** [review]
Quite a mammoth PR! I mostly just looked at the bits that were added and didn't do any validation that the bits that were removed have been copied elsewhere. Looks like a good structure at least, I left only small comments.

**Review by Leaf-Lin — COMMENTED** [review]
Thanks for the work! Added a few comments, nothing major.

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/snapshot-restore/restore-snapshot.asciidoc
Comment (williamrandolph) [inline docs/reference/snapshot-restore/restore-snapshot.asciidoc:28]: Once again I can't find the source for this list, but it looks like it contains this item:

> If you restore a data stream, ensure the cluster contains a matching index template with data stream enabled. Without a matching index template, a data stream can’t roll over or create backing indices.

Is this true if you are restoring global state with your snapshot?

**Review by williamrandolph — COMMENTED** [review]
This looks very good. I had a nice time reading through the preview pages, and really only came up with some questions and potential places to add context.

I looked particularly at sections describing feature state and system indices, and from the perspective of the System Indices project, I think we have what we need here.

File: docs/reference/snapshot-restore/restore-snapshot.asciidoc
Comment (jrodewig) [reply docs/reference/snapshot-restore/restore-snapshot.asciidoc:28]: Thanks for pointing this out. This is intended to cover both cases (existing or restore from cluster state), but it's not clear. I've reworded to:

> Before you restore a data stream, use the [get index template API][0] to ensure the cluster contains a [matching index template][1] with data stream enabled. If no such template exists, you can [create one][1] or [restore a cluster state][2] that contains one. Without a matching index template, a data stream can’t roll over or create backing indices.

[0]: https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-get-template.html
[1]: https://www.elastic.co/guide/en/elasticsearch/reference/master/set-up-a-data-stream.html#create-index-template
[2]: https://www.elastic.co/guide/en/elasticsearch/reference/master/restore-snapshot-api.html#restore-snapshot-api-include-global-state

File: docs/reference/snapshot-restore/register-repository.asciidoc
Comment (jrodewig) [reply docs/reference/snapshot-restore/register-repository.asciidoc]: Elasticsearch Service is our preferred term for our Cloud SaaS. However, I'll add a link to the trail/sign-in/console so it's bit more clear.

File: docs/reference/snapshot-restore/register-repository.asciidoc
Comment (jrodewig) [reply docs/reference/snapshot-restore/register-repository.asciidoc:65]: A goal of this PR is to merge the Elasticsearch and Kibana docs to give users a "one-stop shop" for everything related to snapshot and restore. https://github.com/elastic/kibana/pull/114836 will remove the current Kibana docs once this is merged.

File: docs/reference/snapshot-restore/index.asciidoc
Comment (jrodewig) [reply docs/reference/snapshot-restore/index.asciidoc:37]: This is great feedback. I include some more information about aliases in https://github.com/elastic/elasticsearch/pull/79081/commits/cdefa2a67ce7e89a7705c8113f244b21d932effe.

File: docs/reference/snapshot-restore/register-repository.asciidoc
Comment (jrodewig) [reply docs/reference/snapshot-restore/register-repository.asciidoc]: Thanks for pointing this out. I pushed https://github.com/elastic/elasticsearch/pull/79081/commits/521eec971781effcdbbf0a771ee8e0e5b6326d81 to add some more context and make this a bit more clear from the start.

**Comment by jrodewig** [comment]
Thank you very much for the review and insightful feedback @williamrandolph.

**Comment by jrodewig** [comment]
@Leaf-Lin Do you mind giving this another look and approving if you don't have any objections.

I think we're ready to merge this in.

**Review by williamrandolph — APPROVED** [review]
Thanks for addressing my comments! This looks really good, from my dev's perspective.

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
Going to go ahead and merge this in. Thanks for the reviews, everyone!

**Pull request merged**
